sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -187,6 +187,12 @@ class JDBCOptions(
 
   // An option to allow/disallow pushing down predicate into JDBC data source
   val pushDownPredicate = parameters.getOrElse(JDBC_PUSHDOWN_PREDICATE, "true").toBoolean
+
+  // The local path of user's keytab file, which is assumed to be pre-uploaded to all nodes either
+  // by --files option of spark-submit or manually
+  val keytab = parameters.getOrElse(JDBC_KEYTAB, null)
```

Comment:
So, this can be tricky. If you upload the keytab with `--files`, you don't necessarily know its path. With YARN you will, since historically the YARN backend dumps all files in the executor's CWD, but for other backends that may not be true, and the location may change for each executor in the same application.

I think you can solve that by using `SparkFiles.get` when this option contains just a file name, for example, and document that giving just a file name means you're distributing the keytab with `--files`.

Reply:
> So, this can be tricky. If you upload the keytab with --files, you don't necessarily know its path.

Good point, changing.